e2e/test/scenarios/dashboard/tabs.cy.spec.js

```diff
@@ -78,6 +78,72 @@ describe("scenarios > dashboard > tabs", () => {
       cy.findByText("Our analytics").should("be.visible");
     });
   });
+
+  it("should only fetch cards on the current tab", () => {
+    visitDashboardAndCreateTab({ dashboardId: 1, save: false });
+
+    // Add card to second tab
+    cy.icon("pencil").click();
+    openQuestionsSidebar();
+    sidebar().within(() => {
+      cy.findByText("Orders, Count").click();
+    });
+    saveDashboard();
+
+    cy.intercept(
+      "POST",
+      `/api/dashboard/1/dashcard/1/card/1/query`,
+      cy.spy().as("firstTabQuery"),
+    );
+    cy.intercept(
+      "POST",
+      `/api/dashboard/1/dashcard/2/card/2/query`,
+      cy.spy().as("secondTabQuery"),
+    );
+
+    // Visit first tab and confirm only first card was queried
+    visitDashboard(1, { params: { tab: 1 } });
+    cy.get("@firstTabQuery").should("have.been.calledOnce");
+    cy.get("@secondTabQuery").should("not.have.been.called");
+
+    // Visit second tab and confirm only second card was queried
+    cy.findByRole("tab", { name: "Tab 2" }).click();
+    cy.get("@firstTabQuery").should("have.been.calledOnce");
+    cy.get("@secondTabQuery").should("have.been.calledOnce");
+
+    // Go back to first tab, expect no additional queries
+    cy.findByRole("tab", { name: "Tab 1" }).click();
+    cy.get("@firstTabQuery").should("have.been.calledOnce");
+    cy.get("@secondTabQuery").should("have.been.calledOnce");
+
+    // Go to public dashboard
+    cy.request("PUT", "/api/setting/enable-public-sharing", { value: true });
+    cy.request("POST", `/api/dashboard/1/public_link`).then(
+      ({ body: { uuid } }) => {
+        cy.intercept(
+          "GET",
+          `/api/public/dashboard/${uuid}/dashcard/1/card/1?parameters=%5B%5D`,
+          cy.spy().as("publicFirstTabQuery"),
+        );
+        cy.intercept(
+          "GET",
+          `/api/public/dashboard/${uuid}/dashcard/2/card/2?parameters=%5B%5D`,
+          cy.spy().as("publicSecondTabQuery"),
+        );
+
+        cy.visit(`public/dashboard/${uuid}`);
+      },
+    );
+
+    // Check first tab requests
+    cy.get("@publicFirstTabQuery").should("have.been.calledOnce");
+    cy.get("@publicSecondTabQuery").should("not.have.been.called");
+
+    // Visit second tab and confirm only second card was queried
+    cy.findByRole("tab", { name: "Tab 2" }).click();
+    cy.get("@publicFirstTabQuery").should("have.been.calledOnce");
+    cy.get("@publicSecondTabQuery").should("have.been.calledOnce");
+  });
```

Comment:
I am not testing that the first request is cached like we did for in-app dashboards because currently there is a bug with dashcard query caching in public dashboards.

Caching relies on the json_query field being returned in the BE response to the dashcard query (see [data_fetching.js line 297](https://github.com/metabase/metabase/blob/4323f01cfa0950954d5ae9dc24bca133977067e7/frontend/src/metabase/dashboard/actions/data-fetching.js#L284)). However, currently `json_query` is always empty for public dashboards.

I think since this is an existing bug it's worth tracking in a separate issue and fixing in a separate PR that doesn't block this one.

Reply:
Totally makes sense not to block the PR and fix this in a separate PR 👍